### ISSUE TITLE
Error when attempting inference from Mistral - Arc A770

### ISSUE THREAD

**Issue by ElliottDyson** [intel-analytics/BigDL#9353]
Hello there,

I wanted to preface this by saying that I am running in a Conda environment for python 3.9.18 within Ubuntu 23.10 inside of visual studio code, the command line for attempting inference from the Mistral LLM is correctly set as to operate for the correct Conda environment. I also have the correct wheels for the following packages installed from http://developer.intel.com/ipex-whl-stable-xpu:
intel-extension-for-pytorch 2.0.110+xpu
torch 2.0.1a0+cxx11.abi
torchvision 0.15.2a0+cxx11.abi
oneccl-bind-pt 2.0.100+gpu

As well as bigdl-llm installed via the following command:
pip install --pre --upgrade bigdl-llm[xpu] -f https://developer.intel.com/ipex-whl-stable-xpu
and accelerate version 0.23.0

I was then prompted that cpuinfo could not be imported, so installed py-cpuinfo, resolving the problem.

This resulted in the following  terminal output:

python ./mistral-generate.py --prompt 'What is AI?'
/home/elliott/mambaforge/envs/BigDL-LLM-ARC/lib/python3.9/site-packages/torchvision/io/image.py:13: UserWarning: Failed to load image Python extension: ''If you don't plan on using image functionality from `torchvision.io`, you can ignore this warning. Otherwise, there might be something wrong with your environment. Did you have `libjpeg` or `libpng` installed before building `torchvision` from source?
  warn(
Loading checkpoint shards: 100%|█| 2/2 [00:36<00:0
Special tokens have been added in the vocabulary, make sure the associated word embeddings are fine-tuned or trained.
The attention mask and the pad token id were not set. As a consequence, you may observe unexpected behavior. Please pass your input's `attention_mask` to obtain reliable results.
Setting `pad_token_id` to `eos_token_id`:2 for open-end generation.
Segmentation fault (core dumped)

It's calling mistral-generate.py instead of generate.py because I am also going to trial inferencing llama-2, so the file was renamed to avoid a naming conflict of both files being labelled as generate.py

I hope you will be able to help me try and solve this issue.

Many thanks,
Elliott Dyson

**Comment by jason-dai** [comment]
@yangw1234 please take a look

**Comment by ElliottDyson** [comment]
My apologies, I've solved the issue, however it is rather an odd one to do with versioning that might want to be looked into.

After running the installation process of bigdl-llm again I attempted a new run. During the re-installation process of bigdl-llm I had noticed that that it was installing some other versions for things like accelerate, transformers, etc. I had then tried running the code again, getting an error about mistral, so I attempted reinstalling transformers version 4.34.0 since it seemed that might have been this issue, it was and the code worked flawlessly.

Also, some other issues in terms of installing, it may be worth noting that in order to get everything to this stage, those packages I had to install from https://developer.intel.com/ipex-whl-stable-xpu, I had to download the appropriate wheels followed by running "pip install *.whl" to install them.

Thought I would mention these in case it could help others.

Thank you

**Comment by ElliottDyson** [comment]
Given this is unlikely going to be a standalone case since it's versioning and packaging issues in a new conda environment; I was thinking it may be worth updating the installation guide until the versioning is fixed so that others running these scripts for the first time also don't run into any struggles.

**Comment by jason-dai** [comment]
Yes, we need to somehow document this @hkvision

**Comment by hkvision** [comment]
> Yes, we need to somehow document this @hkvision

Sure.

**Comment by hkvision** [comment]
1. Shall we add `py-cpuinfo` into dependencies? @rnwang04 
2. https://github.com/intel-analytics/BigDL/tree/main/python/llm/example/CPU/HF-Transformers-AutoModels/Model/mistral In the document for Mistral, installing transformers 4.34.0 is already there. Do we need to add it elsewhere? @jason-dai 
But it is a bit strange, since from our side the wrong transformers version would result in the error like`KeyError mistral`, but not core dump...

**Comment by hkvision** [comment]
> Also, some other issues in terms of installing, it may be worth noting that in order to get everything to this stage, those packages I had to install from https://developer.intel.com/ipex-whl-stable-xpu, I had to download the appropriate wheels followed by running "pip install *.whl" to install them.

This issue may be the same as https://github.com/intel-analytics/BigDL/issues/8979 ?

**Comment by rnwang04** [comment]
> 1. Shall we add `py-cpuinfo` into dependencies? @rnwang04
I think `py-cpuinfo` is already in dependencies (https://github.com/intel-analytics/BigDL/blob/main/python/llm/setup.py#L270)

**Comment by hkvision** [comment]
@plusbang will help to highlight the transformers version in mistral doc :)

**Comment by plusbang** [comment]
Now the transformers version requirement is highlighted in mistral examples.
